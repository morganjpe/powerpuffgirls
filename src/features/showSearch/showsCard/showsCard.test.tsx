import { screen } from '@testing-library/react';
import ShowsCard from './showsCard';
import pretty from 'pretty';

import renderWithBrowser from '../../../testutils';

const IMAGE_PATH =
  'https://www.screwfix.com/images/CAT145B/assets/gfx/145b_summer2_makita_twin_hw_lg.png';
describe('<ShowsCard /> component', () => {
  it('should display the show name', () => {
    const { container } = renderWithBrowser(
      <ShowsCard showId={9999} title="Show Title" image={IMAGE_PATH} />
    );
    screen.getByText(/show title/i);
    const image = screen.getAllByAltText(/show title/i);

    image.forEach((img) => {
      expect(img).toHaveAttribute('src', IMAGE_PATH);
    });

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
