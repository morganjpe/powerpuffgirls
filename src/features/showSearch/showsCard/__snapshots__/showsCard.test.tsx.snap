// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowsCard /> component should display the show name 1`] = `
"<div><a href=\\"/show/9999\\">
    <div class=\\"sc-gtsrHT jMaFal\\"><img src=\\"https://www.screwfix.com/images/CAT145B/assets/gfx/145b_summer2_makita_twin_hw_lg.png\\" alt=\\"Show Title\\" class=\\"sc-dlnjwi ejJcht\\"><img class=\\"main\\" tw=\\"rounded\\" src=\\"https://www.screwfix.com/images/CAT145B/assets/gfx/145b_summer2_makita_twin_hw_lg.png\\" alt=\\"Show Title\\"></div>
    <h4>Show Title</h4>
  </a></div>"
`;
